extension/src/popup/views/AccountHistory/index.tsx

```diff
@@ -76,14 +76,16 @@ export const AccountHistory = () => {
   const [detailViewProps, setDetailViewProps] = useState(
     defaultDetailViewProps,
   );
-  const [isLoading, setIsLoading] = useState(false);
 
   const stellarExpertUrl = getStellarExpertUrl(networkDetails);
 
-  const isAccountHistoryLoading =
-    historySegments === null ||
-    accountBalanceStatus === ActionStatus.IDLE ||
-    accountBalanceStatus === ActionStatus.PENDING;
+  useEffect(() => {
+    const getData = async () => {
+      await fetchData();
+    };
+    getData();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
If we use `fetchData` as a dep of the `useEffect` it triggers another load of the data. React officially[ recommends using an empty array ](https://react.dev/reference/react/useEffect#passing-an-empty-dependency-array)to run effects only on mount.

I think we should remove this lint rule but if we really don't want to we can also use the loading state to step around the extra call, but we won't be able to avoid the extra render.

Reply:
I'm fine using the empty array as userEffect dependency, I think it makes sense for cases where we need to run the code only once on mount 👍 

I'm on the fence about removing the lint rule, it could be useful in other cases where it's important to add more deps

Reply:
+1

Reply:
Agreed - let's keep the lint rule and just disable when we explicitly just want to run on mount